astropy/coordinates/distances.py

```diff
@@ -63,34 +62,40 @@ class Distance(u.SpecificTypeQuantity):
     ndmin : int, optional
         See `~astropy.units.Quantity`.
     allow_negative : bool, optional
-        Whether to allow negative distances (which are possible is some
-        cosmologies).  Default: ``False``.
+        Whether to allow negative distances (which are possible in some
+        cosmologies).  Default: `False`.
 
     Raises
     ------
     `~astropy.units.UnitsError`
-        If the ``unit`` is not a distance.
+        If the ``unit`` is not a length unit.
     ValueError
         If value specified is less than 0 and ``allow_negative=False``.
 
-        If ``z`` is provided with a ``unit`` or ``cosmology`` is provided
-        when ``z`` is *not* given, or ``value`` is given as well as ``z``.
+        If ``cosmology`` is provided when ``z`` is *not* given.
 
-        If none of ``value``, ``z``, ``distmod``, or ``parallax`` were given.
+        If either none or more than one of ``value``, ``z``, ``distmod``,
+        or ``parallax`` were given.
 
 
     Examples
     --------
     >>> from astropy import units as u
-    >>> from astropy.cosmology import WMAP5, WMAP7
-    >>> d1 = Distance(10, u.Mpc)
-    >>> d2 = Distance(40, unit=u.au)
-    >>> d3 = Distance(value=5, unit=u.kpc)
-    >>> d4 = Distance(z=0.23)
-    >>> d5 = Distance(z=0.23, cosmology=WMAP5)
-    >>> d6 = Distance(distmod=24.47)
-    >>> d7 = Distance(Distance(10 * u.Mpc))
-    >>> d8 = Distance(parallax=21.34*u.mas)
+    >>> from astropy.cosmology import WMAP5
+    >>> Distance(10, u.Mpc)
+    <Distance 10. Mpc>
+    >>> Distance(40*u.pc, unit=u.kpc)
+    <Distance 0.04 kpc>
+    >>> Distance(z=0.23)                      # doctest: +FLOAT_CMP
+    <Distance 1184.01657566 Mpc>
+    >>> Distance(z=0.23, cosmology=WMAP5)     # doctest: +FLOAT_CMP
+    <Distance 1147.78831918 Mpc>
+    >>> Distance(distmod=24.47)               # doctest: +FLOAT_CMP
```

Comment:
Apparently it is possible to specify a distance modulus without a unit. It might be better to be stricter about enforcing a unit here, but even if it is then making the change is beyond the scope of this pull request.

Reply:
I see that it just assumes a default unit of `u.mag`. I think that is probably not worth changing. Though my sense would be to use `24.47*u.mag` in the example, as it is more explicit.

Reply:
I have added the unit to the example.